core/dbt/parser/schema_yaml_readers.py

```diff
@@ -595,8 +596,14 @@ def parse_semantic_model(self, unparsed: UnparsedSemanticModel):
             config=config,
             unrendered_config=unrendered_config,
             group=config.group,
+            meta=unparsed.meta,
         )
 
+        # If we have meta in the config, copy to node level, for backwards
+        # compatibility with earlier node-only config.
+        if "meta" in config and config["meta"]:
+            parsed.meta = config["meta"]
```

Comment:
Why do we have to support the legacy way for a new node?

Reply:
@QMalcolm specifically because of [this comment](https://github.com/dbt-labs/dbt-core/issues/8511#:~:text=If%20we%20can%27t%20then%2C%20during%20serialization%2C%20copy%20meta%20to%20be%20a%20top%2Dlevel%20attribute%20of%20semantic_models%2C%20as%20we%20do%20with%20other%20node%20types%20%E2%80%94%C2%A0that%20would%20be%20a%20bit%20inconsistent%20for%20metadata%20consumers%2C%20but%20it%27s%20definitely%20not%20a%20show%2Dstopper.).

Reply:
I've pinged @jtcohen6 for clarity on this. I read that comment to mean we've left the top level `meta` attribute on existing nodes to support backwards compatibility, but I now see that it can also be read to mean we need to add it to all new nodes for backwards compatibility.

Reply:
@emmyoop @QMalcolm I'm happy with whatever's easier here.

Strictly, we only _need_ it for existing node types — so for all new node types (including semantic models), we only _need_ to support it nested within `config`.